lib/qesapdeployment.pm

```diff
@@ -944,6 +949,368 @@ sub qesap_az_get_peering_name {
+    Return 0 (false) if delete-transit-gateway-vpc-attachment fails or the gateway does not become inactive before the timeout
+
+=over 2
+
+=item B<NAME> - Prefix for the Tag Name of transit-gateway-vpc-attachment
+
+=item B<TIMEOUT> - default is 5 mins
+
+=back
+=cut
+
+sub qesap_aws_delete_transit_gateway_vpc_attachment {
+    my (%args) = @_;
+    croak 'Missing mandatory name argument' unless $args{name};
+    $args{timeout} //= bmwqemu::scale_timeout(300);
+
+    my $res = qesap_aws_get_transit_gateway_vpc_attachment(
+        name => $args{name});
+    # Here [0] suppose that only one of them match 'name'
+    my $transit_gateway_attachment_id = $res->[0]->{TransitGatewayAttachmentId};
+    return 0 unless $transit_gateway_attachment_id;
+
+    my $cmd = join(' ', 'aws ec2 delete-transit-gateway-vpc-attachment',
+        '--transit-gateway-attachment-id', $transit_gateway_attachment_id);
+    script_run($cmd);
+
+    my $state = 'none';
+    my $duration;
+    my $start_time = time();
+    while ((($duration = time() - $start_time) < $args{timeout}) && ($state !~ m/deleted/)) {
+        sleep 5;
+        $res = qesap_aws_get_transit_gateway_vpc_attachment(
+            transit_gateway_attach_id => $transit_gateway_attachment_id);
+        $state = $res->[0]->{State};
+    }
+    return $duration < $args{timeout};
+}
+
+=head3 qesap_aws_add_route_to_tgw
+    Adding the route to the transit gateway to the routing table in refhost VPC
+
+=over 3
+
+=item B<RTABLE_ID> - Routing table ID
+
+=item B<TARGET_IP_NET> - Target IP network to be added to the Routing table eg. 192.168.11.0/16
+
+=item B<TRANSIT_GATEWAY_ID> - ID of the target Transit gateway (IBS Mirror)
+
+=back
+=cut
+
+sub qesap_aws_add_route_to_tgw {
+    my (%args) = @_;
+    foreach (qw(rtable_id target_ip_net trans_gw_id)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+
+    my $cmd = join(' ',
+        'aws ec2 create-route',
+        '--route-table-id', $args{rtable_id},
+        '--destination-cidr-block', $args{target_ip_net},
+        '--transit-gateway-id', $args{trans_gw_id},
+        '--output text');
+    script_run($cmd);
+}
+
+=head3 qesap_aws_filter_query
+
+    Generic function to compose a aws cli command with:
+      - `aws ec2` something
+      - use both `filter` and `query`
+      - has text output
+
+=cut
+
+sub qesap_aws_filter_query {
+    my (%args) = @_;
+    foreach (qw(cmd filter query)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+
+    my $cmd = join(' ', 'aws ec2', $args{cmd},
+        '--filters', $args{filter},
+        '--query', $args{query},
+        '--output text');
+    return script_output($cmd);
+}
+
+=head3 qesap_aws_get_mirror_tg
+
+    Return the Transient Gateway ID of the IBS Mirror
+
+=cut
+
+sub qesap_aws_get_mirror_tg {
+    return qesap_aws_filter_query(
+        cmd => 'describe-transit-gateways',
+        filter => '"Name=tag-key,Values=Project" "Name=tag-value,Values=IBS Mirror"',
```

Comment:
String `IBS Mirror` should not be here but should come from a variable. Exactly like we did for ResourceGroup in Azure peering, it come from `IBSM_RG`